### DISCUSSION TITLE
Specify check suite when creating a checkrun

### DISCUSSION THREAD

**Discussion by jwgmeligmeyling** [community/community#24616] [Apps, API and Webhooks]
<p>When creating a check run through the API, a check suite is newly created unless a check suite for the requesting application has already been made, in which case that check suite seems to be reused.</p>
<p>This leads to issues however: due to Github Actions workflows, and each Workflow Run (workflow/event) having its own check suite, multiple check suites may exist for a given commit SHA. As a result, annotations generated in a Github action during a step executed in the “pull request” event, end up in the “workflow” for the "push’ event.</p>
<p>Through hoops, it is possible to find the check run ID during a Github Action: one can look up the Workflow Run by the run_id and then extract the Check Suite ID from the checkSuiteUrl. However, there is no way to specify the Check Suite ID when creating a new check run (its ignored in the request) and neither there seems to be any dedicated endpoint for creating check runs for a particular check suite.</p>
<p>It would be great if it were possible to more easily find the corresponding Check Suite for a Workflow Run, and if it were possible to create Check Runs for this particular Check Suite rather than have one picked “randomly” for you.</p>
<p>See also the issue thread <a href="https://github.com/jwgmeligmeyling/pmd-github-action/issues/4" rel="nofollow noopener">https://github.com/jwgmeligmeyling/pmd-github-action/issues/4</a> and proof of concept code in <a href="https://github.com/jwgmeligmeyling/pmd-github-action/pull/3" rel="nofollow noopener">https://github.com/jwgmeligmeyling/pmd-github-action/pull/3</a></p>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="jwgmeligmeyling" data-post="1" data-topic="118380">
<div class="title">
<div class="quote-controls"></div>
 jwgmeligmeyling:</div>
<blockquote>
<p>It would be great if it were possible to more easily find the corresponding Check Suite for a Workflow Run, and if it were possible to create Check Runs for this particular Check Suite rather than have one picked “randomly” for you.</p>
</blockquote>
</aside>
<p><a class="mention" href="/jwgmeligmeyling">@jwgmeligmeyling</a> - Welcome back to the Support Community and thanks for this feedback! :bowing_man:</p>
<p>After reviewing your particular case, I’d like to confirm here that it is not currently possible to create a check run associated with a particular check suite with the <code>GitHub Actions</code>-provided <code>GITHUB_TOKEN</code>. I acknowledge that it is something that would be valuable given your use case.</p>
<p>We’re always working to improve GitHub and the GitHub Support Community, and we consider every suggestion we receive. Would you mind <a href="https://support.github.com/contact/feedback">submitting this through our official product feedback form</a> so that our product team can track your request? That’s the best place to share requests like these in consideration for future iterations of GitHub features, including Actions.</p>

**Reply by carlspring** [reply]
@francisfuzz ,

Ping? Please, see the comments below.

**Reply by ghost** [reply]
@francisfuzz 
Ping?! Please, see the comments below!

**Reply by samus-aran** [reply]
Hi @ruben-hoenle-fnt 👋 

Thanks for sharing and we appreciate your patience. You can check our [public roadmap](https://github.com/github/roadmap/projects/1) to see the kinds of work we are focusing on for the coming year. Smaller fixes won’t be listed, but are often included when work is happening adjacent to it.

While it has been a while since we have updated on this thread, the absence of an update is simply that - there has been no update because there is so far nothing to report. That doesn't mean it isn't being worked on - it might simply be that we're unable to share details about work that isn't ready for release.

As a GitHub user, the most you can do is to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention.

**Comment by jwgmeligmeyling** [comment]
<p>Thanks for your response! I’ve submitted the product feedback form.</p>

**Reply by Akashh2151** [reply]
It seems you're facing challenges with managing check suites and check runs in GitHub Actions workflows, especially when dealing with multiple workflows and check suites for a given commit SHA. Unfortunately, as of my last knowledge update in September 2021, GitHub's API and workflow behavior doesn't directly address these concerns. However, there are some strategies and workarounds you can consider:

Workflow Run Name and Context: When defining your GitHub Actions workflow, you can include a name attribute for each job or step. This can help you differentiate between different runs of the same workflow, especially if you're using multiple workflows for different events. Additionally, you can use environment variables, context, and outputs to pass information between different workflow runs.

Custom Annotations: When creating annotations in your GitHub Actions workflow, you can provide a path attribute to specify the file path to which the annotation corresponds. This way, you can provide more context to the annotations and help differentiate them when viewing them in different contexts.

GitHub Action Output: You can use the set-output command in your workflow steps to define outputs that can be used by subsequent steps or workflows. This can be useful to capture relevant information (such as check suite ID) and pass it to other parts of your workflow.

GitHub App API Access: If you have a GitHub App and need more control over check runs and check suites, you might consider exploring the GitHub App REST API or GraphQL API. These APIs might provide more flexibility for managing check suites and runs, but be aware that they can introduce additional complexity.

Feedback and Feature Requests: If your use case requires more granular control over check suites and runs, consider providing feedback to GitHub through their official support channels or GitHub Discussions. They might be interested in hearing about your use case and considering improvements to the workflow and API behavior.

Community and Third-Party Solutions: Sometimes, the GitHub community creates tools or actions to address specific use cases. It's possible that there might be third-party actions or tools that help with managing check suites and runs more effectively in complex workflows.

As the GitHub platform and ecosystem are continually evolving, it's recommended to stay updated with the latest documentation, releases, and announcements. If you're facing specific limitations that significantly affect your workflow, it's a good idea to communicate your requirements to GitHub support or explore any new features they might introduce in the future.

**Reply by bewuethr** [reply]
Is it really useful to copy-paste ChatGPT answers here ("last knowledge update in September 2021")?

**Reply by queenofcorgis** [reply]
@Akashh2151 While we encourage non-AI answers to questions in community, we also love experimenting and using new technologies. We ask if you do use ChatGPT or similar programs, please disclose what software and model version number you used to generate the response you used in your reply in order to promote transparency in the community.

ChatGPT answers in a tone that sounds like a tech support professional and while we support trying out this software, it is important to verify that they are solutions before posting otherwise it may mislead users into thinking that they are receiving official responses when they are not.

Per ChatGPT's website, the program does have limitations like:
- "May occasionally generate incorrect information"
- "May occasionally produce harmful instructions"
- "Limited knowledge of world and events after 2021"

**Reply by jwgmeligmeyling** [reply]
So instead of reporting this junk as spam we encourage it and ask for the version number? This whole thread is getting more insane by the day.

**Comment by EnricoMi** [comment]
<p>Is this product feedback submitted by <a class="mention" href="/jwgmeligmeyling">@jwgmeligmeyling</a> publicly visible? What is the best way to track progress of this issue?</p>

**Comment by jwgmeligmeyling** [comment]
<p>No I don’t think its publicly available. I didn’t receive any response as well. I’ve filed another support ticket, maybe that gets things going.</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/enricomi">@EnricoMi</a> - Great question! Because product feedback submitted through <a href="https://support.github.com/contact/feedback">this form</a> isn’t publicly visible, it’s not possible to track its progress. However, it may be worth sharing that our team <a href="https://github.community/t/github-roadmap-faq/124978">announced the GitHub public roadmap</a> to give everyone more transparency into what each of our teams are planning and to help everyone understand our product vision. It’s <a href="https://github.com/github/roadmap/projects/1">publicly viewable</a> if you’re interested in following what our team is working on next, though it doesn’t currently show any feedback submitted from that form. I hope this helps!</p>

**Comment by jonico** [comment]
<p>It might be worth trying out to use the token of a separate GitHub App when creating the check run. Here is an action that retrieves a temporary token from a GitHub App created just for this purpose: <a href="https://github.com/peter-murray/workflow-application-token-action" rel="noopener nofollow ugc">https://github.com/peter-murray/workflow-application-token-action</a></p>

**Comment by NasAmin** [comment]
<p>Yes I have the same problem.<br>
When you have multiple workflows triggered on a pull_request event and one of the workflows is creating check runs using the Create Check API, the check runs end up randomly in one of the workflows.</p>
<p>I have also submitted a suggestion on the feedback form.</p>

**Comment by dobegor** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> is there any update on this issue? I wasn’t able to find it on the roadmap you’ve mentioned.</p>

**Comment by eblot** [comment]
<p>I was expecting to use several runners to perform distinct, unrelated checks, but the results appear in a randomly picked workflow tab. Is there any known workaround for now? Thanks.</p>
<p>(note: it sounds weird to have this thread flagged as “solved” :-))</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/dobegor">@dobegor</a> - I don’t have any updates to share on my part nor am I able to provide any in the future. The public roadmap is still the best place to keep up with upcoming features.</p>

**Comment by lukash** [comment]
<p>Hello <a class="mention" href="/francisfuzz">@francisfuzz</a>, this seems to be a “minor” bug from the roadmap PoV, and is unlikely to ever be listed there. Hence this response from you is not really useful at all, as we don’t even know whether the issue is still being tracked by Github internally or it just “fell through the cracks”.</p>
<p>It’s a shame especially because at least on the surface this seems like it should be an easy fix to actually use the Check Suite ID. For my use case it’s causing this: <a href="https://github.com/ScaCap/action-surefire-report/issues/39" class="inline-onebox" rel="noopener nofollow ugc">Report check item appears under the wrong workflow name · Issue #39 · ScaCap/action-surefire-report · GitHub</a> and again, it’s a shame, because that action allows reasonable and very simple test result reporting via workflow annotations, which is clear and easy to find/read even by outside contributors. There are very few options with regards to test result reporting that don’t require outside resources!</p>
<p>I have submitted another feedback form for this, hopefully it can get some more attention. Thank you.</p>

**Comment by thegecko** [comment]
<p>This issue makes it difficult to locate checks for specific workflows and is serious enough for us to consider moving away from GH actions :frowning:</p>

**Comment by inket** [comment]
<p>This is a <strong>BREAKING</strong> issue for GitHub Actions, and it’s honestly mindblowing that it’s come down to “submit feedback and get no answer”. We have <strong>zero</strong> acknowledgment that our feedback was even seen by anyone working on GitHub Actions.</p>
<p>It’s especially sad since we all know this would be a very easy fix. You can literally put an intern on it they’ll get it done in less than a day.</p>

**Reply by radiomix** [reply]
It reminds me of the  _old_ :window:  behavior _It's not a bug. It's a feature_
![windows3 11](https://user-images.githubusercontent.com/6728633/196645559-1ca7c16a-91d5-44aa-bc16-d82af5fea7b4.jpeg)

**Comment by jwgmeligmeyling** [comment]
<p>Sad to see this was still not communicated to the appropriate folks. It is a trivial fix. A dozen people have requested this feature.</p>
<p>And annotations are practically useless as it is.</p>
<p>Someone made a stinky API decision and successfully managed to sweep it under the carpet. Of course we want to be able to specify to which check run annotations should be added, rather than have one picked <strong>randomly</strong> for us! It’s just common sense.</p>

**Comment by MostefaKamalLala** [comment]
<p>This is seriously needed by my team. We moved to github actions and we are even paying for the service. We cannot have test results, annotations spread all over the place randomly. When will we get an update on the API?</p>

**Comment by andriilatyshev** [comment]
<p>Is any result or progress?</p>

**Comment by zolex** [comment]
<p>We are also facing this issue. It is important for us that we can attach a check to a specific workflow run/check-suite. Can’t be that hard to add a new property run_id or suite_id on the create checks api that it now takes over a year since the first(?) appearance of this issue…</p>
<p>I have just sent a request using the feedback form in the name of a big automotive company. Let’s see if they finally work on that problem…</p>

**Comment by bobleujr** [comment]
<p>Also: <a href="https://github.community/t/associate-a-check-run-with-a-workflow/122540" class="inline-onebox">Associate a Check Run with a Workflow</a></p>

**Comment by IanMoroney** [comment]
<p>If people upvote the following discussion, github might get around to fixing this problem:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/github/feedback/discussions?discussions_q=Allow%2Bspecifying%2Bworkflow%2Brun%2Bwhen%2Bcreating%2Ba%2Bcheckrun%2Bfrom%2Ba%2BGitHub%2Bworkflow">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/github/feedback/discussions?discussions_q=Allow%2Bspecifying%2Bworkflow%2Brun%2Bwhen%2Bcreating%2Ba%2Bcheckrun%2Bfrom%2Ba%2BGitHub%2Bworkflow" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/10791260/181085663-6d0f31d1-88f5-42e8-a797-1d650e4ad3a6.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/0/b0814f55732dcd40863bf1cbe604689e6e11ea9b_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/0/b0814f55732dcd40863bf1cbe604689e6e11ea9b_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/0/b0814f55732dcd40863bf1cbe604689e6e11ea9b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/0/b0814f55732dcd40863bf1cbe604689e6e11ea9b_2_10x10.png"></div>

<h3><a href="https://github.com/github/feedback/discussions?discussions_q=Allow%2Bspecifying%2Bworkflow%2Brun%2Bwhen%2Bcreating%2Ba%2Bcheckrun%2Bfrom%2Ba%2BGitHub%2Bworkflow" target="_blank" rel="noopener nofollow ugc">Discussions · github/feedback</a></h3>

  <p>Public feedback discussions for: GitHub for Mobile, GitHub Discussions, GitHub Codespaces, GitHub Sponsors, GitHub Issues and more! - Discussions · github/feedback</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>The more votes we have, hopefully the higher visibility it’ll get</p>

**Comment by gabsmprocha** [comment]
Any update about this problem?

**Reply by jwgmeligmeyling** [reply]
Not that I am aware of.

**Reply by gabsmprocha** [reply]
oh my god, two years later, sad 😢

**Reply by jwgmeligmeyling** [reply]
Yea I was one of the early adopters of the annotations API back when it was in beta. If it even ever left beta 🤣

**Reply by eblot** [reply]
Update is something like "we just don't care". As with many other features (such as the terrible PR review workflow), the implementation is half-complete, half-debugged, half-usable. The only thing that is not halved is the price per seat.

**Comment by devstrat** [comment]
I guess I want too much: having test results properly attached to the correct workflow while having parallel runs, this technology is too advanced for 2022. We're paying customers btw. 2 years and nothing

**Reply by radiomix** [reply]
:100:  LOL!

**Comment by austinpray-mixpanel** [comment]
You can see how confusing this is for our users. The "flake8" check is a python linting check:

![image](https://user-images.githubusercontent.com/71290498/193842281-195ce0e9-3710-4621-9db3-c404fe1cd7fd.png)
![image](https://user-images.githubusercontent.com/71290498/193842305-f40e0c48-b4f8-4460-8079-82e31a3a7157.png)

**Comment by dobegor** [comment]
@francisfuzz is it possible for Github to re-evaluate the priority of this issue?

**Reply by francisfuzz** [reply]
👋 @dobegor,

While I understand this feature is high priority for you, the Product team takes several factors into consideration when prioritizing features.  The Product team is aware of this feedback as it lives in this discussion, but I can't make any guarantees about this feature's prioritization.

**Reply by sifive-eblot** [reply]
This is not a "Feature" request. It is about fixing a nasty bug that plague an existing feature for over 2 years.

As long as "making things usable" is considered as a "new feature" we can wait forever. This existing feature is deeply broken - randomly linking reports - and the whole point of this issue is not to add something new but making it work...

**Reply by radiomix** [reply]
Thread started ` on Jun 17, 2020` 
Now at `Oct. 19, 2022` *LOL*

Feels like the _old_ MS user experience: Well come back to the world of
![windows3 11](https://user-images.githubusercontent.com/6728633/196646597-5816abfa-a5b1-472b-aefd-7ff501f223d7.jpeg)

**Comment by vstirbu** [comment]
This is the proper way to add feedback in actions, equivalent with the check runs in GitHub Apps: https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/

**Reply by chrispat** [reply]
You can also create notice, warning, and error annotations https://docs.github.com/en/enterprise-server@3.4/actions/using-workflows/workflow-commands-for-github-actions#setting-a-notice-message

**Reply by zolex** [reply]
That does not solve the problem at all. In fact only github themselves can fix it...
What you have linked is another feature

**Reply by chrispat** [reply]
It is a feature that can be used to do most of the same things you would do when creating additional check_runs.  Creating check_runs from an action's workflow for the workflow check_suite isn't something that has ever been supported, in fact you can't use those APIs unless you are a GitHub App.  The only reason it sort of works is Actions happens to be an app under the covers.  From what we have seen there are a couple of reasons users wanted to do it and those are covered by the features listed above.  

We are interested to hear specific scenarios not covered by the features above where you need to use that API from Actions.

**Reply by zolex** [reply]
Well, with the summary you can not have checks that are required to merge a pull request

**Reply by vstirbu** [reply]
Depending on the complexity of your workflow files you can require the workflow to merge. See below the `modelcard-test` is the `name` of the workflow:

![image](https://user-images.githubusercontent.com/1288047/197596373-d036e70f-1204-4b71-b61a-1c80f086bed5.png)

**Reply by zolex** [reply]
> Depending on the complexity of your workflow files you can require the workflow to merge. See below the `modelcard-test` is the `name` of the workflow:
> 
> ![image](https://user-images.githubusercontent.com/1288047/197596373-d036e70f-1204-4b71-b61a-1c80f086bed5.png)

this is what I mean, which is not possible when just putting text in summary, as suggested above ;)

**Reply by vstirbu** [reply]
The workflow in the screenshot has a step that outputs the summary.

**Reply by zolex** [reply]
This whole topic is about programmatically added checks...

**Reply by jwgmeligmeyling** [reply]
Wouldn't call the suggested approach the _proper_ way. It might work fine for your goals though. However, the Checks and Annotations API was announced with quite the buzz (I even remember a mail campaign) for exactly this purpose: https://github.blog/2018-05-07-introducing-checks-api/

**Reply by bbrzoska** [reply]
A single Github Action can create multiple checks. Some can pass, others fail, and others still be neutral in outcome. Failures of some may be set as blocking merges, others not. Using the output summary isn't a solution for this problem.

**Comment by AlexPerathoner** [comment]
Same problem on my side. Definitely a problem, or "missing feature", if you prefer.

**Comment by leviplj** [comment]
Is GH going to fix this or should we give up on the checks?

**Reply by jwgmeligmeyling** [reply]
You should give up. Ive waited for years.

Talked to my GH Enterprise rep about it, they push on sarif uploads trough a Github Advanced Security upsell. Mind you, it can process a dozen files at most before crashing on you. The annotation API actually works better.

**Comment by JustinOhms** [comment]
Well this seems to explain why my check runs keep keep ending up associate with my push action and not the action driven by the PR.  I can't believe this is an issue.  If you cannot control what the check run is actually associated with it's next to useless.

**Reply by sarnold** [reply]
<sigh> I can't believe I'm still reading this...

**Comment by carlspring** [comment]
@francisfuzz , @maxprit , 

Can we raise the severity of this? It's not just a desired feature, it's a total misimplementation of how it should work. As you can see, quite a few other developers feel this way and are affected by it.

What do we need to do in order to get the attention of the Github developers/managers? Is there a issue tracker where this can be reported? How do escalations work?

**Reply by steve-todorov** [reply]
We are also experiencing this issue and it is quite disappointing to see such a rookie mistake being made on such an important feature.
To be honest this feature makes absolutely no sense to even exist or be available to the general public, because practically it doesn't work. 

Nobody ever has just a `hello world` project - people have 1, 2 sometimes 3 workflows that deal with the different phases / stages. 
I really hope this gets more attention.

**Comment by obia5313** [comment]
{
  "total_count": 0,
  "runners": []
}

**Comment by obia5313** [comment]
{
  "total_count": 0,
  "runners": []
}

**Comment by dev-hanz-ops** [comment]
There is a workaround. It's possible to publish the check-run in a dedicated, separate check-suite by using a token for a GitHub app, instead of the regular GITHUB_TOKEN.
As example, we use [this action](https://github.com/EnricoMi/publish-unit-test-result-action/issues/12):

- create a [GitHub App](https://docs.github.com/en/apps/creating-github-apps/creating-github-apps/creating-a-github-app) (permissions e.g. `Read and write access to actions, checks, code, discussions, and pull requests` - possibly too broad for your case)
- create a private key for it and store it in your GH Actions secrets
- install the app to your org / repo
- ```
        - id: app_token
          uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
          if: always()
          env:
            OPENSSL_CONF: /dev/null  # https://github.com/tibdex/github-app-token/issues/54
          with:
            app_id: your-app-id
            private_key: ${{ secrets.PRIVATE_KEY_YOUR_APP }}
  
        - name: Publish Test Results
          uses: EnricoMi/publish-unit-test-result-action@v2
          if: always()
          with:
            # when using the regular GITHUB_TOKEN, the check-run created by this step will be assigned to a
            # random workflow in the GH UI. to prevent this, we can force the check-run to be created in a separate
            # check-suite, which is created automatically if we use the credentials of a GitHub App
            github_token: ${{ steps.app_token.outputs.token }}
  ```

update: your workflow should create a comment like the default one using the github-actions user, as otherwise the action will create one comment per workflow run, instead of updating the first:

```
      - name: create test-results comment if not exists
        if:
          always() && (github.event_name == 'workflow_dispatch' || steps.changes.outputs.src == 'true')
        env:
          GH_ENTERPRISE_TOKEN: ${{ secrets.GITHUB_TOKEN }}    # choose one
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}               # choose one
          GH_HOST: your-enterprise-github.com                 # leave blank if using github.com
        run: |
          set -o xtrace
          short_ref=$(IFS='/' read -r -a REF <<< "$GITHUB_REF" && echo ${REF[2]})
          test_result_comment=$(gh pr view $short_ref --comments \
            --json comments \
            --jq '.comments[]
              | select(.author.login == "github-actions")
              | select(.body | startswith("## Test Results"))')
          [[ -z "$test_result_comment" ]] && gh pr comment $short_ref --body $'## Test Results\nResults for commit ' || true
```

result

![image](https://user-images.githubusercontent.com/97662291/231841736-abeb175c-aaea-48b0-bd8c-8542ca61fba2.png)  
![image](https://user-images.githubusercontent.com/97662291/231847604-200ca364-394e-4390-ba8c-4584d4e17b2d.png)

It comes at the cost of not having the published message as part of the GH Actions run - this would otherwise be randomly assigned.

![image](https://user-images.githubusercontent.com/97662291/231848936-8a26e890-3d45-4b7b-b690-735a85f6ea1b.png)

**Reply by EnricoMi** [reply]
This is the first reliable workaround that I have seen for this issue (though not simple to set up).

**Reply by steve-todorov** [reply]
Yeah, this seems to be a reliable workaround. The only downside is when you navigate to the action run you need to go to the summary and then remember to click the `View more details` link. It's still better than nothing though.

**Reply by austinpray-mixpanel** [reply]
It's better than nothing but definitely would expect some way to associate checks with a given workflow. Concrete example: A "Python Linting" workflow that generates checks for flake8, mypy, black, etc

**Reply by steve-todorov** [reply]
We had multiple workflows which published unit tests / integration tests. We had to create multiple github applications in order to separate these reports (otherwise they were overwritten). Maybe this could work for your case?

**Reply by austinpray-mixpanel** [reply]
We already have GitHub actions workflows that publish checks under a github app. The feedback we have been getting in a large repo is that the UX is bad because it disconnects the job the generates the checks from the checks. Also you have to be super careful multiple github actions (pull_request, push, schedule) writing to the same check suite stomping all over each other.

Conceptually it's a github action that generates checks. Why not be able to colocate the checks with the thing that caused them?

<img src="https://user-images.githubusercontent.com/71290498/233144651-85c35162-d763-462d-a00b-7252f30978dc.png" width="300" alt="github problem">

**Reply by steve-todorov** [reply]
I completely agree. We noticed it as well, but there is simply no "going around it". :disappointed: 

@francisfuzz will get annoyed at some point if we ping him enough times. :)

**Reply by carlspring** [reply]
@steve-todorov 
> @francisfuzz will get annoyed at some point if we ping him enough times. :)

Unless he's muted notifications from this thread... :)

@maxprit,

Could anyone at Github comment on this? It's obviously an issue for quite a few people.

**Reply by EnricoMi** [reply]
Maybe raising individual issues at [Github Support](https://support.github.com/contact/bug-report?tags=hubberfy_actions%2Ccat_github_actions) may get their attention.

**Reply by jwgmeligmeyling** [reply]
I am not sure why the sudden influx on this thread. I appreciate y'alls trust in getting this resolved, but after pushing this issue for three years now I can only recommend to put your time elsewhere

**Reply by Vampire** [reply]
@dev-hanz-ops you might consider updating your comment a bit.
Thanks for the effort, but it is a bit outdated in multiple regards:
- you might want to switch to the official `actions/create-github-app-token` action that exists now for such a security sensitive operation
- it is unclear - at least to me - what `steps.changes.outputs.src` is, as such a step was nowhere mentioned
- if leaving in the comment creation it should be made clear that it is not cross-platform usable as-is
- the whole comment creation can be left out
  - since `v2.8.0` the action does have an `github_token_actor` input that needs to be set to the proper login name of the token, which is used for finding the comment to update
  - as soon as my PR https://github.com/EnricoMi/publish-unit-test-result-action/pull/661 is released even that is not necessary anymore, as it just fixes the proper automatic comment recognition

:-)

**Comment by EnricoMi** [comment]
Github support's response is:

> Using the Checks API to generate check_runs at runtime from an Actions workflow is not a supported use scenario.

The API is meant to be called from a Github app, but not from Githubs very own Github app called Github Actions.

The proper use in Github Actions is to create your own Github app, install that app in your repos, get an installation token and use that with the Checks API (see https://github.com/orgs/community/discussions/24616#discussioncomment-5607870 above).

Very user-friendly.

**Reply by carlspring** [reply]
This discussion sounds like a front row seat at a stand-up comedy show! :grinning:

**Comment by mutantcornholio** [comment]
Just to clear something out: if you have a single workflow that is triggered by serveral different webhook events, you can't have it resulting in a single status on a commit? 

An example: 
* an action is triggered on a `pull_request` event
* then somebody reviews a PR (triggering a `pull_request_review` event)

So the action can't create a commit status in first case, which would be updated in the second?

**Reply by samrees** [reply]
I spent all day on this before I found this thread, trying to merge my status between those two events.

**Comment by alexmaurizio** [comment]
It is now May 2024 and this trivially fixable problem still plagues the whole GitHub Actions ecosystem. 

I'm definitely baffled.

**Comment by bfollonikea** [comment]
Encountered this issue recently. Sad to see this thread was open in 2022 and the problem still exists.

**Comment by mashedkeyboard** [comment]
Just encountered this myself and spent a solid ten minutes staring at the code, the workflow run, and the GitHub support form, before finding this discussion... this seems bizarre! Adding a voice to ask for this to be fixed please.

**Comment by eXorus** [comment]
Same for me, I had to redo all my code to account for this bug.

**Comment by br3akzero** [comment]
+1 experiencing this issue with this awesome eslint action helper https://github.com/ataylorme/eslint-annotate-action/issues/35

**Comment by diegopalat** [comment]
Any news on this topic?

**Comment by radiomix** [comment]
I came across the issue and first looked at this post in 2022. :hourglass_flowing_sand: 

Recently we configured _GitHub Advanced Security_ to trigger `CodeQL` runs. :heavy_check_mark:  

Sadly they and up with the status of our `maven-surefire` check. :x: 

Maybe I need to _ponder_  a workaround :thinking:

**Reply by radiomix** [reply]
Yep: It was Oct *2022* : https://github.com/orgs/community/discussions/24616#discussioncomment-3913327 :hourglass_flowing_sand:

**Comment by mdreano** [comment]
I just found this post, I spent sometime trying to configure my workflow only to finally understand that it is a bug.
Any update would be appreciated although I understand it may not be a major issue and priority for github team.

**Comment by man1207** [comment]
Still not fixed

**Comment by nguyta02** [comment]
do we have any future plan to fix this issue ?

**Comment by tanin47** [comment]
I just also encountered this issue which would be solved by this capability : https://github.com/mikepenz/action-junit-report/issues/1361#issuecomment-2952188074

**Comment by thompsonjared** [comment]
This is also an issue for me, had check runs & code annotations working exactly as I wanted in my deployment pipeline.

Then I enabled automatic dependency submission to take advantage of GitHub dependency scanning...which introduced a 2nd workflow run on the same commit.

End result my deployment workflow check runs mysteriously disappear from the workflow they were created from and instead show up in the automated GitHub security workflow run.

GitHub is 99% of the way towards make this a great feature for actions, and then it's just fumbled on the 1 yard line.... with apparently no progress on the issue for 5 years of comment history.

**Comment by hrkh** [comment]
We're also experiencing this issue where Check Runs created via the API are being linked to the wrong workflow, especially when several workflows run for the same commit. This makes tracking our CI processes more difficult than it should be.

To resolve this, it would be extremely valuable if the API allowed us to specify check_suite_id when creating a Check Run.

**Comment by lewis-jackson-bots** [comment]
Just tripped up on this too after a day of trying to figure out why it wasn't working. We're trying to avoid running ~20 test shards by posting skipped (successful) checks in certain scenarios.

Running 20 jobs for <2 seconds each to get a successful check costs us 20 minutes of GitHub action runner time as the minimum is 1 minute.

**Comment by zuzusik** [comment]
Facing this issue as well. If a pipeline ran more than once for commit, then all the subsequent checks created via API with `GITHUB_TOKEN` are attached to check suite created with the first run. This results in inability to merge PR if check is marked as required, as checks end up being attached to the stale check suite.

**Comment by chrisjenx** [comment]
Don't want to me-too, but chaps this is years old at this point... if you want people to use Actions vs other CI solutions, you need to put effort into dev...

**Comment by JustinOhms** [comment]
At this point maybe someone who actually cares about this should get a job
with GitHub just to fix this one thing.  Because that’s the only way I see
this actually getting fixed at this point. 😆

**Reply by chrisjenx** [reply]
Based on timelines, you could go to college, graduate, join as a IC1. Would be quicker than waiting for them to fix it!

**Reply by Vampire** [reply]
Please consider that each message you send is pinging and reaching every person following this discussion. ;-)

**Reply by radiomix** [reply]
Please consider to hit the  :bell:  `Unsubscribe` button on the top-left of this page.

**Comment by diegopalat** [comment]
❓❓❓ Any news on this topic?  it is about 6 years now.
